lib/msf/base/sessions/pingback.rb

```diff
@@ -0,0 +1,90 @@
+# -*- coding: binary -*-
+require 'msf/base'
+
+module Msf
+module Sessions
+
+###
+#
+# This class provides basic interaction with a command shell on the remote
+# endpoint.  This session is initialized with a stream that will be used
+# as the pipe for reading and writing the command shell.
+#
+###
+class Pingback
+
+  #
+  # This interface supports basic interaction.
+  #
+  include Msf::Session::Basic
+
+  #
+  # Returns the type of session.
+  #
+  def self.type
+    "pingback"
+  end
+
+  def initialize(conn, opts = {})
+    self.platform ||= ""
+    self.arch     ||= ""
+    datastore = opts[:datastore]
+    super
+  end
+
+  def self.create_session(conn, opts = {})
+    uuid_raw = conn.get_once(16, 1)
+    if uuid_raw
+      uuid_string = uuid_raw.each_byte.map { |b| "%02x" % b.to_i() }.join
+      puts "Incoming UUID = #{uuid_string}"
```

Comment:
If this is development output or messaging for the user? If it is for the user consider using the appropriate `print_*` method.